include/caffe/util/db.hpp

```diff
@@ -181,6 +181,87 @@ class LMDB : public DB {
   MDB_dbi mdb_dbi_;
 };
 
+
+#define MAX_BUF 10485760  // max entry size
```

Comment:
It prevents a too large `key_size` or `data_size` read from file, maybe from file corruption.
I thought 10M for a datum is large enough, or maybe should be larger? 100M?